### PR TITLE
Use group_by operator instead of Aggregate in SQL compiler output

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Also, fix instances in the tutorial using Aggregate and replace with group_by.
Interestingly, most changes were already made in the test tutorial.dl.

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
@amytai: can you please review the SQL compiler changes?

**Comment by ryzhyk** [comment]
@mbudiu-vmw, @amytai, please make sure your SQL compiler improvements are documented in the `Unreleased` section of CHANGELOG.

**Comment by mbudiu-vmw** [comment]
They aren't visible to users in any way; we are just avoiding using Aggregate for group_by.
It's not really worth a line in the changelog.
If anything the tutorial fixes are.

**Comment by mbudiu-vmw** [comment]
BTW: @ryzhyk , you may need to review the tutorial changes.

**Comment by ryzhyk** [comment]
> They aren't visible to users in any way; we are just avoiding using Aggregate for group_by.

I meant in general, not just this commit.

> BTW: @ryzhyk , you may need to review the tutorial changes.

Tutorial changes look good, thanks!

File: sql/src/main/java/com/vmware/ddlog/ir/DDlogEApply.java [hunk 1]
```diff
@@ -34,18 +34,31 @@
 public class DDlogEApply extends DDlogExpression {
     private final String func;
     private final List<DDlogExpression> args;
+    // If true the suffix notation with dot is preferred.
+    private boolean suffix;
```
Comment (amytai) [inline sql/src/main/java/com/vmware/ddlog/ir/DDlogEApply.java:38]: Is this used in this PR anywhere? If not, what's an intended use-case?

File: sql/src/main/java/com/vmware/ddlog/ir/DDlogRHSGroupby.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "var " + this.var + " = " + this.rhsProject.toString() +
+                ".group_by((" + String.join(", ", this.rhsGroupByVars) + "))";
+    }
+
+    @Override
+    public boolean compare(DDlogRuleRHS val, IComparePolicy policy) {
```
Comment (amytai) [inline sql/src/main/java/com/vmware/ddlog/ir/DDlogRHSGroupby.java:50]: Why do these nodes need to be Comparable?

**Review by amytai — APPROVED** [review]

File: sql/src/main/java/com/vmware/ddlog/ir/DDlogEApply.java (hunk 1 above)
Comment (mbudiu-vmw) [reply sql/src/main/java/com/vmware/ddlog/ir/DDlogEApply.java:38]: Not yet used, but may be, the result is easier to read.

File: sql/src/main/java/com/vmware/ddlog/ir/DDlogRHSGroupby.java (hunk 2 above)
Comment (mbudiu-vmw) [reply sql/src/main/java/com/vmware/ddlog/ir/DDlogRHSGroupby.java:50]: This is not the standard Java comparison. This was intended to make test comparisons more robust by checking if two DDlog programs are equivalent up to alpha-renaming. Then we would not need to change the test outputs by hand every time a new variable is renamed. But that would first require writing a DDlog parser, and we don't have that yet.

**Comment by mbudiu-vmw** [comment]
Fixes #1068

**Comment by mbudiu-vmw** [comment]
@amytai please review the last commit which hopefully fixes the issue you reported.

**Review by amytai — APPROVED** [review]

**Pull request merged**